unstructured/unstructured.go

```diff
@@ -0,0 +1,103 @@
+package unstructured
+
+import (
+	"errors"
+)
+
+// MissingField is the error returned when the requested field could not
+// be found.
+var MissingField = errors.New("field not found")
+
+// InvalidIndex is the error returned when trying to access an
+// out-of-bound item in an array.
+var InvalidIndex = errors.New("index is out of bound")
+
+// InvalidType is the error returned when trying to access a field from
+// a non-map, or an index from a non-array.
+var InvalidType = errors.New("invalid type")
+
+// Unstructured is the object that let's you manipulate Data.
```

Comment:
What benefits can we get if we don't user the one in api-machinery: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/unstructured.go#L41-L46

Reply:
apimachinery's unstructured type is much more specific than this type.